.github/workflows/snap.yml

```diff
@@ -19,34 +19,55 @@ env:
   DEFAULT_RISK: edge
 
 jobs:
+  # This job is here just to avoid having "greens" in the "snaps" job for 
+  # architectures that aren't really being built and tested
```

Comment:
This dynamically building the `matrix` is clever, but I think a bit tricky to understand/maintain. I wonder if there's a better way to structure this? Maybe we can just skip building the snap altogether on PR branches? Or maybe we can use a `snap-pr.yml` that applies only on `pull_request` and re-`uses` jobs from this file?

Reply:
Y, I mean, the best alternative is to have a PR-specific job. I did this to avoid having duplicated steps, which will be the case with the PR-specific job.

But I can do it. See https://github.com/canonical/pebble/pull/326/commits/5c4517e1996ccb41d267013e749e1923abf0362e